File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.py
# Last updated: 7/14/2026, 2:02:59 PM
class Solution:
    def firstUniqChar(self, s: str) -> int:
        count = {}
        for ch in s:
            count[ch] = count.get(ch, 0) + 1
        for i in range(len(s)):
            if count[s[i]] == 1:
                return i

        return -1

        

File: 387-FirstUniqueCharacterInAString/387-FirstUniqueCharacterInAString.java
// Last updated: 7/14/2026, 2:02:53 PM
class Solution {
    public int firstUniqChar(String s) {
        int []fre=new int[26];
        for(int i=0;i<s.length();i++){
            fre[s.charAt(i)-'a']++;
        }
        for(int i=0;i<s.length();i++){
            if(fre[s.charAt(i)-'a']==1){
                return i;
            }
        }
        return -1;
    }
}